src/Sintegra.php

```diff
@@ -51,38 +53,65 @@ public function get()
      */
     protected function totalize($sintegra)
     {
-        $tot = '';
+        $sintegraArray = explode("\n", $sintegra);
+        $numeroTotalRegistros = count($sintegraArray);
+        $numeroTotalRegistros += 1;
+        $numeroTotalRegistros = str_pad($numeroTotalRegistros, 8, "0", STR_PAD_LEFT);
+        $numeroTotalRegistros = str_pad($numeroTotalRegistros, 53, " ", STR_PAD_RIGHT);
+        $totalRegistros90 = '2';
+
+        $cnpj = substr($sintegraArray[0], 2, 14);
+        $cnpj = str_pad($cnpj, 14, " ", STR_PAD_RIGHT);
+        $ie = substr($sintegraArray[0], 16, 14);
+        $ie = str_pad($ie, 14, " ", STR_PAD_RIGHT);
+
         $keys = [];
-        $aefd = explode("\n", $sintegra);
-        foreach ($aefd as $element) {
-            $param = explode("|", $element);
-            if (!empty($param[1])) {
-                $key = $param[1];
-                if (!empty($keys[$key])) {
-                    $keys[$key] += 1;
-                } else {
-                    $keys[$key] = 1;
+        $totalizador = '';
+        $somatorioPorBloco = [
+            SintegraEnum::REGISTRO_10 => 0,
+            SintegraEnum::REGISTRO_11 => 0,
+            SintegraEnum::REGISTRO_50 => 0,
+            SintegraEnum::REGISTRO_51 => 0,
+            SintegraEnum::REGISTRO_53 => 0,
+            SintegraEnum::REGISTRO_54 => 0,
+            SintegraEnum::REGISTRO_60 => 0,
+            SintegraEnum::REGISTRO_61 => 0,
+            SintegraEnum::REGISTRO_70 => 0,
+            SintegraEnum::REGISTRO_71 => 0,
+            SintegraEnum::REGISTRO_74 => 0,
+            SintegraEnum::REGISTRO_75 => 0,
+            SintegraEnum::REGISTRO_85 => 0,
+            SintegraEnum::REGISTRO_86 => 0,
+            SintegraEnum::REGISTRO_88 => 0,
+        ];
+
+        foreach ($sintegraArray as $element) {
+            $numeroBloco = substr($element,0, 2);
+
+            if (!empty($numeroBloco)) {
+                if (!isset($keys[$numeroBloco])) {
+                    $somatorioPorBloco[$numeroBloco] += 1;
+                    continue;
                 }
+                $somatorioPorBloco[$numeroBloco] = 1;
```

Comment:
Essa parte eu fui meio burro dá para deixar dessa porque o array já tem a chave:
```suggestion
            if (!empty($numeroBloco)) {
                $somatorioPorBloco[$numeroBloco] += 1;
```